src/lib/Microsoft.Health.Fhir.Ingest.Template/HandlerProxyTemplateFactory.cs

```diff
@@ -0,0 +1,59 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Health.Common.Handler;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Ingest.Template
+{
+    public abstract class HandlerProxyTemplateFactory<TInput, TOutput> :
+        ITemplateFactory<TInput, TOutput>,
+        IResponsibilityHandler<TInput, TOutput, IList<string>>
+        where TOutput : class
+    {
+        private readonly IList<string> _templateErrors = new List<string>();
+
+        protected IList<string> TemplateErrors
+        {
+            get { return _templateErrors; }
+        }
+
+        /// <summary>
+        /// Create mapping template for the container of serialzied JSON template.
+        /// Throw exception for template validation errors.
+        /// </summary>
+        /// <param name="jsonTemplate">JSON template container </param>
+        /// <returns>Mapping template object.</returns>
+        public abstract TOutput Create(TInput jsonTemplate);
+
+        /// <summary>
+        /// Create mapping template for the container of serialzied JSON template.
+        /// Return the deserialized mapping object with validation errors.
+        /// </summary>
+        /// <typeparam name="TContext">The type of context container</typeparam>
+        /// <param name="jsonTemplate">JSON template container </param>
+        /// <param name="errors">Template validation errors</param>
+        /// <returns>Mapping template object.</returns>
+        public abstract TOutput Create(TInput jsonTemplate, out IList<string> errors);
+
+        TOutput IResponsibilityHandler<TInput, TOutput, IList<string>>.Evaluate(TInput request) => Create(request);
+
+        TOutput IResponsibilityHandler<TInput, TOutput, IList<string>>.Evaluate(TInput request, out IList<string> errors) => Create(request, out errors);
+
+        protected JsonSerializer GetJsonSerializer()
+        {
+            return JsonSerializer.Create(new JsonSerializerSettings
+            {
+                MissingMemberHandling = MissingMemberHandling.Error,
```

Comment:
I think this is the other way around. It means that if JSON has a property that doesn't have a match in the Class schema, it would fail. https://www.newtonsoft.com/json/help/html/DeserializeMissingMemberHandling.htm

Since this caused confusion to the reading, I'll leave a short comment on it.